### ISSUE TITLE
Code highlighting for PHP nowdoc/heredoc block fails

### ISSUE THREAD

**Issue by valtlait** [adobe/brackets#11376]
Brackets’ code highlighting for PHP nowdoc/heredoc block fails when using
&nbsp; – quotes around the identifier (the snippet 2 in test code)
&nbsp; – white-space between `<<<` and the identifier (” 3 ”)

Highlighting for the code snippet 1 works correctly.

A screenshot (Brackets 1.3):

![brackets-php-highlighting-fails_2](https://cloud.githubusercontent.com/assets/5496284/8635408/05a4d36c-282a-11e5-850b-0275e267d120.png)

Test code:

```php
<?php

// 1
echo <<<EOT
	...
EOT;

// 2
echo <<<'EOT'
	...
EOT;

// 3
echo <<< EOT
	...
EOT;
```

**Comment by mackenza** [comment]
This would need to be fixed in CodeMirror (what Brackets uses for its editor window) as I tried it there and CM has the same issues. I would recommend logging an issue on the CM repo against their PHP mode. If they fix it there, it would be picked up "for free" when Brackets upgrades their CM version.

**Comment by mackenza** [comment]
CM fixed an issue codemirror/codemirror#3340 that fixes your case 2 (which is really a nowdoc not a heredoc). The third case, with the space before the identifier, is still present after I patched using 3340.

Note that codemirror/codemirror#3340 seems to be fixed in their version 5.4.0 but Brackets is currently using v5.3.1 so you could either apply 5.4.0 (or just the patch for that issue) or wait until Brackets picks up CM 5.4.0. 

This issue should be kept open until then imo.

**Comment by valtlait** [comment]
@mackenza, thank you for the information!

**Comment by MarcelGerber** [comment]
Upstream bug filed: codemirror/CodeMirror#3421

**Comment by MarcelGerber** [comment]
Confirmed this works on master, now that we run the latest CM version again.
Closing. You'll get this fix in Release 1.5
